projects/packages/forms/src/contact-form/class-contact-form.php

```diff
@@ -267,6 +294,10 @@ public static function parse( $attributes, $content ) {
 				$attributes['block_template_part'] = $GLOBALS['grunion_block_template_part_id'];
 			}
 		}
+
+		if ( is_singular() ) {
+			add_action( 'admin_bar_menu', array( __CLASS__, 'add_quick_link_to_admin_bar' ), 500 );
```

Comment:
Is there a particular reason for the `500` priority? Are there any dependencies here that influence this? If so, perhaps they could be communicated via an inline comment.

Also, can you explain the reasoning behind adding the hook in this parse function processing a shortcode rather than earlier in the request's lifecycle? Is this due to the potential for shortcodes to be filtered and even potentially removed by those filters?

Reply:
No particular reason for 500 priority.